### PR TITLE
Ft trip history

### PR DESCRIPTION
**Pull request by edgar929**
requester or manager can view trip history.
how to test this:
-- run (npm run dev)
-- in postman use this routes
   *localhost:3000/api/v1/trips/Nairobi/?page=0&limit=2 (to view all trips made by a given location)
   *localhost:3000/api/v1/trips (to view grouped list of all trips made by location)

### CODE REVIEW

**Review by songa1 — COMMENTED** [review]
I think it's fine.  All are working fine and from my view, you are ready to go. And i think about your build status, you can check your barefoot.yml file which is in GitHub/workflows

**Review by Blaiseniyo — APPROVED** [review]

**Comment by Blaiseniyo** [comment]
I think it is good to go

File: src/controllers/viewTripHistory.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+export const getTripHistory = async (req, res, next) => {
+  const decoded = await getDataFromToken(req, res, next);
+  
+    const userid = decoded.id.toString();
+    const offset = req.query.page;
+    const limit = req.query.limit;
+    const location = req.params.location;
+    let query = { userId: userid };
+    findTrip(res, query, location, offset, limit);
```
Comment (bdushimi) [inline src/controllers/viewTripHistory.js:14]: @edgar929 you don't need to pass the res object to findTrip function. FindTrip function should only get parameters/arguments that it only needs to find the trip info. FindTrip function should return the data i.e. trip info and then getTripHistory returns the response to the user. It's not a good idea to keep passing down req & res object. Kindly update elsewhere you've passed the res object down to another function.

File: src/middlewares/sendEmail.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export const sendVerificationEmail = async (req, res, next) => {
     }
 
   } catch (error) {
+    next(error);
```
Comment (bdushimi) [inline src/middlewares/sendEmail.js:29]: @edgar929 you can pass the error object then in the errorHandler function, you can display whatever property of error.

File: src/services/tripHistorySearch.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      return res.status(401).json(err.message);
+    }
+
+};
+
+
+export const displayNumberOfTrips = async (res, query) => {
+    let resultSet1 = [];
+    
+        await db.TravelRequest.findAndCountAll({ where: query })
```
Comment (bdushimi) [inline src/services/tripHistorySearch.js:30]: @edgar929 could you reduce this function complexity, lenght and improve its readability?
It has lots of .thens and ifs besides async & await aren't adequately used.

**Review by bdushimi — CHANGES_REQUESTED** [review]

**Comment by bdushimi** [comment]
@edgar929 Kindly resolve the merge conflicts

**Pull request merged**